bos

```diff
@@ -1051,28 +1051,34 @@ prog
   .option('--external-funding', 'Use external funds for the channel open')
   .option('--give <give_amount>', 'Amount to gift to peer', REPEATABLE)
   .option('--node <node_name>', 'Node to open channels')
+  .option('--opening-node <opening_node>', 'Node to open channels', REPEATABLE)
   .option('--set-fee-rate <ppm>', 'Set forward fee rate to peer', REPEATABLE)
   .option('--type <type>', 'Type of channel (private/public)', REPEATABLE)
   .action((args, options, logger) => {
     return new Promise(async (resolve, reject) => {
       const collect = n => flatten([n]).filter(n => !!n);
+      const openingNodes = n => flatten([n]);
+      const nodes = !!options.openingNode ? openingNodes(options.openingNode).map(String) : [];
+      const lnds = !!nodes.length ? await Promise.all(nodes.map(async node => (await lnd.authenticatedLnd({logger, node: node !== 'true' ? node: options.node})).lnd)) : [];
```

Comment:
ideally there is no logic in this method, or minimal logic, this could potentially be a new helper function to get referenced saved nodes

since getting saved node can fail it should be within the try block

Reply:
Where should this helper function be? A new file in the peers dir or in the lnd dir? 
And call it from openChannels or from bos?

Reply:
either one is reasonable, basically just doing the same thing except not in the bos file